naga/src/front/wgsl/parse/directive/language_extension.rs

```diff
@@ -54,14 +52,22 @@ impl LanguageExtension {
 }
 
 /// A variant of [`LanguageExtension::Implemented`].
-#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq, VariantArray)]
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
+#[cfg_attr(test, derive(strum::VariantArray))]
 pub enum ImplementedLanguageExtension {
     ReadOnlyAndReadWriteStorageTextures,
     Packed4x8IntegerDotProduct,
     PointerCompositeAccess,
 }
 
 impl ImplementedLanguageExtension {
+    /// A slice of all variants of [`ImplementedLanguageExtension`].
+    pub const VARIANTS: &'static [Self] = &[
+        Self::ReadOnlyAndReadWriteStorageTextures,
+        Self::Packed4x8IntegerDotProduct,
+        Self::PointerCompositeAccess,
+    ];
```

Comment:
fun-fact: This is how I originally done it: https://github.com/gfx-rs/wgpu/pull/6814#discussion_r1904968508.

Reply:
As a reviewer for that PR: Yeah, I hadn't considered just validating stuff in `test` profiles. 😅